src/utils.rs

```diff
@@ -120,6 +120,13 @@ where
     Option::<T>::deserialize(de).map(|opt| opt.unwrap_or_else(Default::default))
 }
 
+pub fn ser_int_with_default<S>(value: &Option<u32>, serializer: S) -> Result<S::Ok, S::Error>
```

Comment:
I'm not sure this is the good name.
You are serialiazing a `Option<u32>`, not an `int`